## Exercício 8 
###  Verificar se a distribuição da região de procedência é a mesma nesta amostra. Qual sua conclusão ?

In [23]:
# importa biblioteca pandas
import pandas as pd

# importa biblioteca scipy
import scipy.stats as stats

# local do arquivo
Caminho='/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'


In [15]:
 #Traz o arquivo para o Python
df=pd.read_excel(Caminho)


In [16]:
df.head()

ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out

In [17]:
#Tabela de frequência da variável grau de instrução
freq=df["grau_instrucao"].value_counts()
freq

grau_instrucao
m    203
f    120
s    103
Name: count, dtype: int64

In [18]:
#Teste Chi-Quadrado
stats.chisquare(freq)


Power_divergenceResult(statistic=40.32394366197183, pvalue=1.752939347286805e-09)

In [19]:
#Tabela de frequência da variável n_filhos
freq2=df["n_filhos"].value_counts()
freq2

n_filhos
0    234
2     75
1     61
3     42
5      8
4      6
Name: count, dtype: int64

In [20]:
#Teste Chi-Quadrado
stats.chisquare(freq2)


Power_divergenceResult(statistic=503.09859154929575, pvalue=1.7116482149421455e-106)

In [21]:
#Tabela de frequência da variável grau de instrução
freq=df["reg_proc"].value_counts()
freq

reg_proc
out    148
cap    143
int    135
Name: count, dtype: int64

In [22]:
stats.chisquare(freq)
#aceita a igualdade das proporcoes

Power_divergenceResult(statistic=0.6056338028169014, pvalue=0.7387343452063055)

## Resposta Questao 8 
### Verificar se a distribuição da região de procedência é a mesma nesta amostra. Qual sua conclusão ?

In [24]:
# Importar bibliotecas necessárias
import pandas as pd
import scipy.stats as stats


In [26]:
# Local do arquivo
caminho = '/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'

In [27]:
# Carregar o arquivo para o DataFrame
df = pd.read_excel(caminho)

In [28]:
# Visualizar as primeiras linhas do DataFrame
print(df.head())

   ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out


In [29]:
# Tabela de frequência da variável reg_proc
freq_reg_proc = df['reg_proc'].value_counts()
print("Frequência da variável reg_proc:")
print(freq_reg_proc)


Frequência da variável reg_proc:
reg_proc
out    148
cap    143
int    135
Name: count, dtype: int64


In [30]:
# Realizar o teste Chi-Quadrado
chi2_stat, p_val = stats.chisquare(freq_reg_proc)
print("\nResultados do Teste Chi-Quadrado:")
print(f"Estatística Chi-Quadrado: {chi2_stat}")
print(f"Valor p: {p_val}")


Resultados do Teste Chi-Quadrado:
Estatística Chi-Quadrado: 0.6056338028169014
Valor p: 0.7387343452063055


In [31]:
# Interpretação e Conclusão
if p_val < 0.05:
    print("Rejeitamos a hipótese nula. A distribuição da região de procedência não é a mesma na amostra.")
else:
    print("Não rejeitamos a hipótese nula. A distribuição da região de procedência é a mesma na amostra.")

Não rejeitamos a hipótese nula. A distribuição da região de procedência é a mesma na amostra.


Interpretação e Conclusão
## Resultados do Teste Chi-Quadrado:

### Interpretação:
- **Estatística Chi-Quadrado (statistic):** 0.6056338028169014
- **Valor p (pvalue):** 0.7387343452063055

O valor p é maior que o nível de significância comum (0.05). Isso indica que não há evidência suficiente para rejeitar a hipótese nula de que a distribuição da região de procedência é a mesma na amostra.

### Conclusão:
Com base nos resultados do teste Chi-Quadrado, não rejeitamos a hipótese nula. Portanto, concluímos que a distribuição da região de procedência é a mesma nesta amostra.

